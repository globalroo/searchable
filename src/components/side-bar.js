import React from "react";

import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";
import { useImageLoader } from "../hooks/image-loader";

export const SideBar = ({ poster_path, title, children }) => {
	const [image] = useImageLoader(getSmallPosterImage(poster_path));
	return (
		<aside className="sidebar">
			<img src={image} alt={title} />
			{children}
		</aside>
	);
};
