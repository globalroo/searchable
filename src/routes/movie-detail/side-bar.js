import React from "react";
import { useImageLoader } from "src/hooks/image-loader";
import { getSmallPosterImage } from "src/tmdb-service/tmdb-config";

export const SideBar = ({ poster_path, title }) => {
	const [image] = useImageLoader(getSmallPosterImage(poster_path));
	return (
		<aside className="sidebar">
			<img src={image} alt={title} />
		</aside>
	);
};
