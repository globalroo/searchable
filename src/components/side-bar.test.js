import React from "react";
import { render, cleanup } from "react-testing-library";
import { SideBar } from "./side-bar";
import "jest-dom/extend-expect";

describe("SideBar Renders as expected", () => {
	afterEach(cleanup);
	it("Renders a component with a navigate back wrapper", () => {
		const { container } = render(
			<SideBar poster_path="test_image.png" title="image holder">
				<p>Test children render</p>
			</SideBar>
		);
		expect(container).toHaveTextContent("Test children render");
		expect(container).toMatchSnapshot();
	});
});
