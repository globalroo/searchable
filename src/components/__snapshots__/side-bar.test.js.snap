// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar Renders as expected Renders a component with a navigate back wrapper 1`] = `
<div>
  <aside
    class="sidebar"
  >
    <img
      alt="image holder"
      src="http://image.tmdb.org/t/p/w300/test_image.png"
    />
    <p>
      Test children render
    </p>
  </aside>
</div>
`;
